Remotion/Data/DomainObjects.UnitTests/Linq/IntegrationTests/TableValuedParameterIntegrationTest.cs

```diff
@@ -0,0 +1,239 @@
+        new ObjectID(typeof(Order), new Guid("F7607CBC-AB34-465C-B282-0531D51F3B04")));
+  }
+
+  [Test]
+  public void Contains_UsingTableValuedParameter_DateTime ()
+  {
+    var deliveryDates = new[] { new DateTime(2005,2,1), new DateTime(2013,3,7) };
+
+    var orders =
+        from o in QueryFactory.CreateLinqQuery<Order>()
+        where deliveryDates.Contains(o.DeliveryDate)
+        select o;
+
+    CheckQueryResult(
+        orders,
+        new ObjectID(typeof(Order), new Guid("F4016F41-F4E4-429E-B8D1-659C8C480A67")),
+        new ObjectID(typeof(Order), new Guid("F7607CBC-AB34-465C-B282-0531D51F3B04")));
+  }
+
+  [Test]
+  public void Contains_UsingTableValuedParameter_String ()
+  {
+    var products = new[] { "Hitchhiker's guide", "Rubik's Cube" };
+
+    var orderItems =
+        from o in QueryFactory.CreateLinqQuery<OrderItem>()
+        where products.Contains(o.Product)
+        select o;
+
+    CheckQueryResult(
+        orderItems,
+        new ObjectID(typeof(OrderItem), new Guid("DC20E0EB-4B55-4F23-89CF-6D6478F96D3B")),
+        new ObjectID(typeof(OrderItem), new Guid("386D99F9-B0BA-4C55-8F22-BF194A3D745A")));
+  }
+
+  [Test]
+  public void Contains_UsingTableValuedParameter_Decimal ()
+  {
+    var prices = new[] { 0.01m, 1m };
+
+    var products =
+        from o in QueryFactory.CreateLinqQuery<Product>()
+        where prices.Contains(o.Price)
+        select o;
+
+    CheckQueryResult(
+        products,
+        new ObjectID(typeof(Product), new Guid("8DD65EF7-BDDA-433E-B081-725B4D53317D")),
+        new ObjectID(typeof(Product), new Guid("BA684594-CF77-4009-A010-B70B30396A01")));
+  }
+
+  [Test]
+  public void Contains_UsingTableValuedParameter_Enum ()
+  {
+    var values = new[] { ClassWithAllDataTypes.EnumType.Value1 };
+
+    var objects =
+        from o in QueryFactory.CreateLinqQuery<ClassWithAllDataTypes>()
+        where values.Contains(o.EnumProperty)
+        select o;
+
+    CheckQueryResult(
+        objects,
+        new ObjectID(typeof(ClassWithAllDataTypes), new Guid("3F647D79-0CAF-4A53-BAA7-A56831F8CE2D")));
+  }
+
+  [Test]
+  public void Contains_UsingTableValuedParameter_Double ()
+  {
+    var values = new[] { 987654.321 };
+
+    var objects =
+        from o in QueryFactory.CreateLinqQuery<ClassWithAllDataTypes>()
+        where values.Contains(o.DoubleProperty)
+        select o;
+
+    CheckQueryResult(
+        objects,
+        new ObjectID(typeof(ClassWithAllDataTypes), new Guid("3F647D79-0CAF-4A53-BAA7-A56831F8CE2D")));
+  }
+
+  [Test]
+  public void Contains_UsingTableValuedParameter_ExtensibleEnum ()
+  {
+    var values = new[] { Color.Values.Blue() };
+
+    var objects =
+        from o in QueryFactory.CreateLinqQuery<ClassWithAllDataTypes>()
+        where values.Contains(o.ExtensibleEnumProperty)
+        select o;
+
+    CheckQueryResult(
+        objects,
+        new ObjectID(typeof(ClassWithAllDataTypes), new Guid("583EC716-8443-4B55-92BF-09F7C8768529")));
+  }
+
```

Comment:
I think we should have a string example as well. and possilby one with binary, if there is a good (=short) value available in the test data..